main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+func main() {
+	// 環境変数を取得
+	localPath, gcpKey, projectId, bucketName, storageClass, duration, bucketNum, webhookId, webhookSecret := EnvVarLoad()
+	log.Print("Env-vars successfully loaded")
+
+	log.Printf("Backin' up files from \"%s\" to \"%s\" on gcp Storage...", localPath, projectId)
+	startTime := time.Now()
+
+	// クライアントを建てる
+	client, err := CreateClient(gcpKey, projectId)
+	if err != nil {
+		log.Print("Error: failed to load create client")
+		panic(err)
+	}
+	defer client.Close()
+
+	// bucketName + 月(mod n) をバケット名とし、bucketNameに再代入
+	t := &startTime
+	bucketName = fmt.Sprintf("%s-%d", bucketName, t.Month()%time.Month(bucketNum))
+
+	// バケットを作成
+	bucket, err := CreateBucket(*client, projectId, storageClass, duration, bucketName)
+	if err != nil {
+		if err.Error() == "Error 409" {
+			log.Print("Bucket with the same name exists. Objects will be overwritten.")
+		} else {
+			log.Print("Error: failed to create bucket")
+			panic(err)
+		}
+	}
+
+	// バケットへディレクトリをコピー
+	objectNum, err, errs := CopyDirectory(*bucket, localPath)
+	if err != nil {
+		log.Print("Error: failed to load local directory")
```

Comment:
ここでエラーがかえったからと言って、「ディレクトリのコピー」の処理の中の「ディレクトリのロード」のエラーであることは関数の中身を知らない限りわからないと思うので、

```suggestion
		log.Print("Error: failed to copy directory:", err)
```

のがよいです
こうすれば関数の実装が変わっても不整合が発生しません
(ほかの箇所も`err`を出力したほうがいいです)

Reply:
それでもいいんですけど、ファイルコピー中にエラーが発生した場合でもpanicすることになるんですが、それでいいんですか？
Conohaのオブジェクトストレージのバックアップの方ではいくつかエラーが発生しても残りのファイルはバックアップしておこう的な発想だと思ったんですが
やるとしたらerrsにappendして返すってのは考えはしましたが

Reply:
> ファイルコピー中にエラーが発生した場合でもpanicすることになる

エラーログの内容を変えることはロジックには影響しないと思います
(`err`と`errs`をまとめるべきっていう意図ではないです)

Reply:
あ～なるほど
この後他の要因でpanicしたいと思った時でも大丈夫なようにってことですね、ようやく理解しました
理解力無くてすみません

Reply:
直ってなかった
>(ほかの箇所もerrを出力したほうがいいです)

に関して
panicでエラー出してますけどprintでも出したほうがいいですかね？その方が見やすい？

Reply:
panicの中身って別の文字列の方がいいんでしょうか？もしくは全部panicの中にまとめちゃった方がいい？

Reply:
あんまり考えてなかったんですが、`panic`側にまとめたほうがいいですね
そうすれば`stderr`に流れるので(`print`だと`stdout`にいくはず)

Reply:
fmt.Errorf使ってpanic側にまとめることにします